camelot/admin/action/list_action.py

```diff
@@ -1040,3 +1050,108 @@ def remove_only(self):
         return True
 
 remove_selection = RemoveSelection()
+
+class EditProfileMixin:
+
+    database_error_title = ugettext('Could not connect to database, please check host and port')
+    media_location_not_readable_title = ugettext('Media location path is not accessible')
+    media_location_not_writeable_title = ugettext('Media location path is not writeable')
+
+    def validate_profile(self, profile):
+        from camelot.view import action_steps
+        # Validate database settings
+        yield action_steps.UpdateProgress(text=ugettext('Verifying database settings'))
+        try:
+            #if not profile.dialect:
+            #    raise RuntimeError('No database dialect selected')
+            engine = profile.create_engine()
+            connection = engine.raw_connection()
+            cursor = connection.cursor()
+            cursor.close()
+            connection.close()
+        except Exception as e:
+            exception_box = action_steps.MessageBox( title = self.database_error_title,
+                                                     text = _('Verify driver, host and port or contact your system administrator'),
+                                                     standard_buttons = [QtWidgets.QMessageBox.StandardButton.Ok] )
+            exception_box.informative_text = str(e)
+            yield exception_box
+            return False
+        # Validate media location
+        info = QtCore.QFileInfo(profile.media_location)
+        if not info.isReadable(): # or not profile.media_location:
+            yield action_steps.MessageBox( title = self.media_location_not_readable_title,
+                                                   text = _('Verify that the path exists and it is readable or contact your system administrator'),
+                                                   standard_buttons = [QtWidgets.QMessageBox.StandardButton.Ok] )
+            return False
+        if not info.isWritable():
+            yield action_steps.MessageBox( title = self.media_location_not_writeable_title,
+                                                   text = _('Verify that the path is writeable or contact your system administrator'),
+                                                   standard_buttons = [QtWidgets.QMessageBox.StandardButton.Ok] )
+            return False
+        # Success
+        return True
+
+class AddNewProfile(AddNewObject, EditProfileMixin):
+
+    name = 'add_new_profile'
+
+    def edit_object(self, new_object, model_context, admin):
+        from camelot.view import action_steps
+        from camelot.core.profile import Profile
+        app_admin = admin.get_application_admin()
+        profile_admin = app_admin.get_related_admin(Profile)
+        while True:
+            try:
+                yield action_steps.ChangeObject(new_object, profile_admin)
```

Comment:
di goed hoor, maar deze mag in principe wel eenvoudiger, dwz we doen de add to proxy pas als het valide is etc. ...